producer-kafka/src/main/java/io/stargate/producer/kafka/configuration/DefaultConfigLoader.java

```diff
@@ -32,8 +35,37 @@
   public CDCKafkaConfig loadConfig(Map<String, Object> options) {
     String topicPrefixName = getTopicPrefixName(options);
     Map<String, Object> kafkaProducerSettings = filterKafkaProducerSettings(options);
+    MetricsConfig metricsConfig = loadMetricsConfig(options);
+    registerMetricsIfEnabled(kafkaProducerSettings, metricsConfig);
     String schemaRegistryUrl = getSchemaRegistryUrl(kafkaProducerSettings);
-    return new CDCKafkaConfig(topicPrefixName, schemaRegistryUrl, kafkaProducerSettings);
+    return new CDCKafkaConfig(
+        topicPrefixName, schemaRegistryUrl, kafkaProducerSettings, metricsConfig);
+  }
+
+  private void registerMetricsIfEnabled(
+      Map<String, Object> kafkaProducerSettings, MetricsConfig metricsConfig) {
+    if (metricsConfig.isMetricsEnabled()) {
+      kafkaProducerSettings.put(
+          CommonClientConfigs.METRIC_REPORTER_CLASSES_CONFIG,
+          DropwizardMetricsReporter.class.getName());
```

Comment:
This is the main integration point. The `DropwizardMetricsReporter` is extending the `org.apache.kafka.common.metrics.MetricsReporter;` plugging into the Kafka producer lifecycle. To see the integration, visit https://github.com/dropwizard/dropwizard-kafka/blob/master/src/main/java/io/dropwizard/kafka/metrics/DropwizardMetricsReporter.java.